front/app/containers/Admin/reporting/components/Tabs.tsx

```diff
@@ -85,11 +86,20 @@ const DashboardTabs = ({ reportBuilderEnabled, children }: Props) => {
 };
 
 const DashboardTabsWrapper = ({ children }: { children: React.ReactNode }) => {
-  const reportBuilderEnabled = useFeatureFlag({ name: 'report_builder' });
-  if (reportBuilderEnabled === undefined) return null;
+  const appConfig = useAppConfiguration();
+  if (isNilOrError(appConfig)) {
+    return null;
+  }
+
+  // If the feature is not allowed by the pricing plan,
+  // we want to show an info message about this, which requires seeing the tab
+  const showReportBuilderTab = showFeatureFlagPaymentTease(
+    appConfig,
+    'report_builder'
+  );
```

Comment:
Sorry, I want to approve and not discuss any longer but it still doesn't make sense to me. 🙂 

If I read this code, I understand it as "we want to show the tab if we want to show the payment tease". This is not a correct understanding of mine because we also want to show it if the feature is simply (allowed &) enabled, so the function name doesn't describe that. 

If we look at the function's implementation this becomes clearer: 
1) it checks the condition we have for showing the tease, namely the feature is not allowed (per the customer's pricing plan).
2) we are, however, also checking if the feature is enabled. Which is basically a feature flag check that we've added to this function, but has nothing to do with showing the tease. It's separated from that. 

So what we could do is:
`const showReportBuilderTab = showFeatureFlagPaymentTease(...) || isFeatureEnabled;`

Essentially take that enabled check out and use showFeatureFlagPaymentTease  only to check the tease condition (feature is not allowed) and add the feature flag outside of it.

But then I don't see why we still need `showFeatureFlagPaymentTease`, because we have the `onlyCheckAllowed` option in `useFeatureFlag` that does exactly this. So we could just have:

```
const showPlanUpgradeTease = !isFeatureAllowed;
const showReportBuilderTab = showPlanUpgradeTease || isFeatureEnabeld;
```

I understand it's convenient to put this in 1 function, but this smells like premature optimization to me because we don't know if we'll be checking a feature flag in other places. Also, I wouldn't know what this function should be called to clearly convey it's checking both the "show plan upgrade" condition as well as the feature flag check.

Maybe I'm just not getting why the current abstraction is good. 😬 

In any case, I think it would be good to have a second opinion of @IvaKop. This is a new pattern we're introducing in the front-end so it's a good practice to hear here 2 cents.

Reply:
I agree with your suggestion, Brent! It does make it simpler to understand (at least in my mind) and we have the necessary abstractions in place to implement the feature as it stands. It might feel a little counter-intuitive to use `useFeatureFlag` in two different ways for the same feature to derive the necessary values but, in this case, I think it makes sense (as long as we name the constants that come out of these hooks clearly) 👍 It's also what we are doing in the abstracted `showFeatureFlagPaymentTease` in any case.

To be clear, there might be value in abstracting this pattern moving forward but let's maybe first make sure we are aware of all the use cases (usually comes with repetition) and we really know what we need + we really need it in many places. As a general rule of thumb (not universal, of course) I wait until I have copy-pasted the same thing at least 3 times before thinking about generalising it in some way (and even then it might not be necessary).

Reply:
Noted, thank all. Let me make the change